Lib/sndhdr.py

```diff
@@ -49,20 +49,28 @@
 SndHeaders.sampwidth.__doc__ = ("""Either the sample size in bits or
 'A' for A-LAW or 'U' for u-LAW.""")
 
+
 def what(filename):
     """Guess the type of a sound file."""
-    res = whathdr(filename)
-    return res
+    try:
+        res = whathdr(filename)
+        return res
+    except (FileNotFoundError) as e:
+        return None
 
 
 def whathdr(filename):
     """Recognize sound headers."""
-    with open(filename, 'rb') as f:
-        h = f.read(512)
-        for tf in tests:
-            res = tf(h, f)
-            if res:
-                return SndHeaders(*res)
+    try:
+        with open(filename, 'rb') as f:
+            h = f.read(512)
+            for tf in tests:
+                res = tf(h, f)
+                if res:
+                    return SndHeaders(*res)
+        return None
+
+    except(TypeError) as e:
```

Comment:
Same here, ``except TypeError``

Reply:
@tiran will do. definitely was silly to create `e` and not use it. thanks